presto-main/src/main/java/com/facebook/presto/operator/aggregation/builder/SpillableHashAggregationBuilder.java

```diff
@@ -189,10 +189,10 @@ private boolean shouldMergeWithMemory(long memorySize)
     @Override
     public void close()
     {
-        if (hashAggregationBuilder != null) {
-            hashAggregationBuilder.close();
-        }
         try (Closer closer = Closer.create()) {
+            if (hashAggregationBuilder != null) {
+                closer.register(hashAggregationBuilder::close);
```

Comment:
why `hashAggregationBuilder` is not `Optional` as others?

Reply:
Because, in fact, it's not optional. It's always non-null. See however https://github.com/prestodb/presto/pull/8728#discussion_r133591506 for explanation why i retain if-not-null check in `close`.

Reply:
So IMO instead of using IFs to check if it is not null, you should be using cheskState. What do you think?

Reply:
> Since `close()` is quite sensitive operation, I'd prefer to err on the side of defensive programming.